Suggest a standard set of terms to reduce confusion across docs (#222)
appwrite/docs: README.md

| Console Users | Refers to users that are registered to have access to the Appwrite Console. Not a proper noun, use as "a console user" or "the console user". Differenciate this clearly from users created through a Client SDK. |
| Client SDK    | Refers to SDKs used by Web, Flutter, Android, and Apple applications. Use as a proper noun, use "a Client SDK" or "the Client SDKs", not client-side SDKs.                                                        |
| Server SDK    | Refers to SDKs used by backend languages like Java, Node.js, or PHP. Use as a proper noun, use "a Server SDK" or "the Server SDKs", not server-side SDKs.                                                         |
| Developer     | Refers to Appwrite developers that use Appwrite to create applications.                                                                                                                                           |
| End User      | Refers to end users of applications with an Appwrite backend. This doesn't refer to developers that interact with Appwrite directly.   |

> | Appwrite Instance | Refers to a single, self-hosted deployment of Appwrite   |

Author: ## Need Feedback

Do we want to call it a deployment of Appwrite, an instance of Appwrite? What language makes the most sense?

> | Application   | Refers to the application built by the Appwrite developer. Can be referred to as web app, mobile app, flutter app, etc. |

Author: ## Need Feedback

Do we want to call it a deployment of Appwrite, an instance of Appwrite? What language makes the most sense?

Reviewer: I think a cluster might make more sense, not 100% sure.

Author: I will post this on this week's state of the docs to get more feedback.

| Method                | Refers to SDK methods like the method `account.get()`. This helps differentiate SDK methods and Appwite Functions in written language.                                                                                                                             |
| Parameters            | Refers to parameters in an SDK method or API definition. Use in cases where we speak about needs, such as "takes parameters x, y, and z" or "requires parameters x and y".                                                                                         |

> | Arguments             | Refers to arguments passed to CLI, SDK method, or REST API. Use when the argument has a concrete value and when we refer to invoking a method, such as "pass in arguments x and y, or "call account.create() with the argument "unique()" to generate a random id. |

Reviewer: These 3 seems very generic and not Appwrite specific

#### Authentication
| **Term**               | **Suggested Usage.**                                                                                                                                                                                                                                          |
|------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Authentication Service | The Authentication service encompases the Account API, Users API, and Teams API.                                                                                                                                                                              |

Author: ### Need Feedback

Do we want to call services
1. Appwrite Authentication
2. Authentication Service (entire proper noun, which is proper if we're consistent)
3. Authentication service (where Authentication, the service, is the sole proper noun)

> | Authentication Service | The Authentication service encompases the Account API, Users API, and Teams API.                                                                                                                                                                              |

Author: ### Need Feedback

Do we want to call services
1. Appwrite Authentication
2. Authentication Service (entire proper noun, which is proper if we're consistent)
3. Authentication service (where Authentication, the service, is the sole proper noun)

Reviewer: I would vote for 2

Author: We should rename the button. The word Users doesn't make a ton of sense.

| Databases Service | Write as the "Database service" (as a proper noun). The Databases service is referred to as "Databases service", not "Database service".                                                              |
| Database          | As of 0.15, the Databases service allows you to create multiple, individual databases. Use "a database" or "databases" as uncapitalized nouns, which differentiates them from the "Database service". |
| Collection        | Use "collection" as a noun (no capitalization), use as "a collection" or "collections".  Avoid using similar terms like "table".                                                                      |
| Document          | Use "document" as a noun (no capitalization), use as "a document" or "documents".  Avoid using similar terms like "entry" or "row".                                                                   |
| Attribute         | Use "attribute" as a noun (no capitalization), use as "a attribute" or "attributes".  Avoid using similar terms like "column" or "key/value".                                                         |

> | Index             | Use "index" as a noun (no capitalization), use as "a index" or "indexes". Use "indexes" instead of indices in a DB related context.                                                                   |

Reviewer: I think we should also mention `Adapter`. This one is used across Databases, Storage, Error Logging. Sometimes used for OAuth providers.

| Bucket          | Write as "a bucket" or "buckets", not a proper noun.                                                                                          |
| File            | Write as "a file" or "files", not a proper noun.                                                                                              |

#### Functions
| **Term**              | **Suggested Usage.**                                                                                                                                                        |
|-----------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Functions Service     | Write as the "Functions service"(as a proper noun). Not "Function service" or "Cloud Functions".                                                                            |
| Function (individual) | When we mention a specific function and not the service as a whole, use "a function" or "functions". Not "a cloud function" or "a Function".                                |

> | Create                | When we "create" a function, it refers to the process of using "Add Function" in console or using `functions.create()` method. No code is uploaded.                         |

Author: Nice, we'll change this when the new console rolls out.

| Deploy/Deployment     | When we say we "deploy" a function or create a new deployment, this is when we upload code through the Create Deployment button or endpoint.                                |
| Activate Deployment   | When we enable a particular version of a function, we say we "activate" the deployment.                                                                                     |

> | Runtimes              | When we refer to a runtime like `node.js 15.5`, call it a "Node.js runtime" or "function runtime". Avoid similar terms like "runtime environment" or "function environment" |

Reviewer: Missing Execution

# Documentation Guidelines

## 🚀 Contributing

## Before Proposing Additions

## Style Guidelines

### Code Examples

### Notices

### Screenshots

### Terminology
Appwrite has many services and features with many ways to express the same concepts. To avoid confusion, this section suggests a standard set of terms used across Appwrite documentation to describe features and concepts.

> #### General

Reviewer 2: Let's maybe add:
- Usage stats
- Audit logs

Author: I don't have any experience with these, what terms do you suggest under them?

> | **Term**      | **Suggested Usage.**                                                                                                                                                                                              |

Reviewer 2: ```suggestion
| **Term**      | **Suggested Usage**                                                                                                                                                                                              |
```

Same everywhere

| Projects      | Each Appwrite instance can have many Appwrite Projects. Use "Appwrite Projects" as a proper noun when referring to the concept, use project or projects when referring to specific projects.                      |
| Console       | Refers to the the GUI. Can also be referred to as the "Appwrite Console" to distinguish from a Mac, Linux, or Windows machine's console.                                                                          |
| Console Users | Refers to users that are registered to have access to the Appwrite Console. Not a proper noun, use as "a console user" or "the console user". Differenciate this clearly from users created through a Client SDK. |
| Client SDK    | Refers to SDKs used by Web, Flutter, Android, and Apple applications. Use as a proper noun, use "a Client SDK" or "the Client SDKs", not client-side SDKs.                                                        |
| Server SDK    | Refers to SDKs used by backend languages like Java, Node.js, or PHP. Use as a proper noun, use "a Server SDK" or "the Server SDKs", not server-side SDKs.                                                         |

> | Adaptor       | Refers to interfaces used to connect Appwrite with third party technologies. Adaptors are found for OAuth, Databases, Storage, and error logging. When referring to adaptors, use specific adaptor names, such as "an OAuth adaptor" or "a Storage adaptor" to avoid confusion.|

Author: We will stick with Adatper, Adaptor is technically correct in English, but I think less used. I won't dig into it further because it really doesn't matter :P

https://grammarist.com/spelling/adapter-adaptor/

| Developer     | Refers to Appwrite developers that use Appwrite to create applications.                                                                                                                                           |
| End User      | Refers to end users of applications with an Appwrite backend. This doesn't refer to developers that interact with Appwrite directly.   |

> | Appwrite Cluster | Refers to a single, self-hosted deployment of Appwrite. Use as "an Appwrite Cluster" or "Appwrite Clusters".|

Author: @stnguyen90 This was my original proposal XD

> | Appwrite Cluster | Refers to a single, self-hosted deployment of Appwrite. Use as "an Appwrite Cluster" or "Appwrite Clusters".|

Reviewer 3: I agree. Cluster doesn't make sense unless we have several machines. I vote for `Appwrite Instance` too

| Application   | Refers to the application built by the Appwrite developer. Can be referred to as web app, mobile app, flutter app, etc. |

> | Method                | Refers to SDK methods like the method `account.get()`. This helps differentiate SDK methods and Appwite Functions in written language.                                                                                                                             |

Author: This is fine. They would be in fact called methods in most textbooks.

> | Method                | Refers to SDK methods like the method `account.get()`. This helps differentiate SDK methods and Appwite Functions in written language.                                                                                                                             |

Author: 👀 I think this can work. TBH, I'm only concerned about confusing a function from the Functions Service, when we talk about this and the SDK.

| Parameters            | Refers to parameters in an SDK method or API definition. Use in cases where we speak about needs, such as "takes parameters x, y, and z" or "requires parameters x and y".                                                                                         |

> | Arguments             | Refers to arguments passed to CLI, SDK method, or REST API. Use when the argument has a concrete value and when we refer to invoking a method, such as "pass in arguments x and y, or "call account.create() with the argument "unique()" to generate a random id. |

Author: They're actually distinctly different in meaning. You define a function with a set of parameters. But when the function is called, what you pass in are actually called arguments. It's different, misusing them will be "incorrect" and confusing depending on who's using it.

> | Arguments             | Refers to arguments passed to CLI, SDK method, or REST API. Use when the argument has a concrete value and when we refer to invoking a method, such as "pass in arguments x and y, or "call account.create() with the argument "unique()" to generate a random id. |

Reviewer 3: Agree with your explanation @gewenyu99 . Those are the technically correct terms 👍

> #### Authentication

Reviewer 2: Let's maybe add:
-Membership

Author: Question for you. There's a few concepts that are actually confusing right now:

- Team membership vs role:member 
- Team roles vs role:<role> (like role:guest)

These terms are very similar in context and meaning, but aren't the same. Should we consider definite terms for these?

CC: @eldadfux

| Account API            | The Account API is an API under the Authentication service that is used by Client SDKs. Avoid "Account service" to avoid confusing developers that it is different from the "Users API" because it might imply they are separate services.                    |
| Users API              | The User API is an API under the Authentication service that is used by Server SDKs. Avoid "Users service" to avoid confusing developers that it is accessing users different from the "Account API" because it might imply they are separate services.       |
| Teams API              | The Teams API is an API under the Authentication service that is used to organize created accounts. Avoid "Teams service", because it is not a stand alone service.                                                                                           |
| Authentication Method  | An authentication method refers to ways to authenticate a user like email, phone, and JWT. Not a proper noun, avoid terms like auth method or login method.                                                                                                   |

> | OAuth2 Providers       | An OAuth2 provider refers to ways to authenticate users using accounts from service providers like AWS, Facebook, or Microsoft. Avoid using simplified terms like OAuth provider and provider to avoid confusion (there are multiple types of providers now). |

Reviewer 2: Not sure if AWS example is relevant. Let's do something tech companies uses for oauth, like GitHub.

Reviewer 2: I would actually put GitHub as first example. Then even Google maybe.

| Sign In/Log In      | Verb form of creating a user session. Use "create session" where relevant, and use sign in/log in to refer to actions on a frontend application.                            |
| Sign Out/Log Out    | Verb form of deleting a user session.  Use "delete session" where relevant, and use sign out/log out to refer to actions on a frontend application.                         |
| Login/Signin        | Noun form referring to the information used to sign in/log in. For example, "to log in, provide valid login information".                                                   |
| Register            | Use "create account" where relevant, and use register/sign up to refer to actions on a frontend application.                                                                |
 

> #### Database

Reviewer 2: Let's maybe add:
- Permissions
- Permission level

| Index             | Use "index" as a noun (no capitalization), use as "a index" or "indexes". Use "indexes" instead of indices in a DB related context.                                                                   |

#### Storage

| **Term**        | **Suggested Usage.**                                                                                                                          |
|-----------------|-----------------------------------------------------------------------------------------------------------------------------------------------|
| Storage Service | Write as the "Storage service" (as a proper noun).                                                                                            |
| Bucket          | Write as "a bucket" or "buckets", not a proper noun.                                                                                          |
| File            | Write as "a file" or "files", not a proper noun.                                                                                              |

> #### Functions

Reviewer 2: Let's maybe add:
- Variable (variables?)

# Documentation Guidelines

## 🚀 Contributing

## Before Proposing Additions

## Style Guidelines

### Code Examples

### Notices

### Screenshots

### Terminology
Appwrite has many services and features with many ways to express the same concepts. To avoid confusion, this section suggests a standard set of terms used across Appwrite documentation to describe features and concepts.
#### General
| **Term**      | **Suggested Usage**                                                                                                                                                                                              |
|---------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Appwrite      | Written as "Appwrite" not "AppWrite" or "appwrite".                      |

Author: I like it

| Projects      | Each Appwrite instance can have many Appwrite Projects. Use "Appwrite Projects" as a proper noun when referring to the concept, use project or projects when referring to specific projects.                      |

> | Web Console       | Refers to the the web GUI. Can also be referred to as the "Appwrite Web Console" to distinguish from a Mac, Linux, or Windows machine's console or Chrome's developer console.                                                                          |

Reviewer 3: It makes more sense to use simpler terms and call it the `Appwrite Dashboard`. As you have rightly pointed out, using `console` makes us need to disambiguate between the OS specific consoles / chrome developer console.

| Create                | When we "create" a function, it refers to the process of using "Add Function" in console or using `functions.create()` method. No code is uploaded.                         |
| Deploy/Deployment     | When we say we "deploy" a function or create a new deployment, this is when we upload code through the Create Deployment button or endpoint.                                |
| Activate Deployment   | When we enable a particular version of a function, we say we "activate" the deployment.                                                                                     |
| Runtimes              | When we refer to a runtime like `node.js 15.5`, call it a "Node.js runtime" or "function runtime". Avoid similar terms like "runtime environment" or "function environment" |
| Execution            | When a function is run, triggered by an event, or triggerd by a CRON job, an "execution" is created. In documentation, refer to "running" a function as "creating an execution".|

> | Variables            | When a function is run, a set of variables are passed in through the `req` object. Some are generated by the runtime, some are defined under the settigs of a function. These are referred to as "variables" and not "environment variables".|

Reviewer 4: typo and rather than:

```suggestion
| Variables            | When a function is run, a set of variables are passed in through the `req` object. Some are generated by the runtime, some are defined under the settings of a function. These are referred to as "variables" rather than "environment variables".|
```

It's a little confusing because they end up in the `req.env` variable.
